jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookieStore.java

```diff
@@ -0,0 +1,388 @@
+                domain = domain.toLowerCase(Locale.ENGLISH);
+            uriDomain = uriDomain.toLowerCase(Locale.ENGLISH);
+            if (cookieDomain != null)
+                cookieDomain = cookieDomain.toLowerCase(Locale.ENGLISH);
+            if (cookieDomain == null || cookieDomain.endsWith("."))
+            {
+                // RFC 6265, section 4.1.2.3.
+                // No cookie domain -> cookie sent only to origin server.
+                return uriDomain.equals(domain);
+            }
+            if (cookieDomain.startsWith("."))
+                cookieDomain = cookieDomain.substring(1);
+            if (uriDomain.endsWith(cookieDomain))
+            {
+                // The domain is the same as, or a subdomain of, the cookie domain.
+                int beforeMatch = uriDomain.length() - cookieDomain.length() - 1;
+                // Domains are the same.
+                if (beforeMatch == -1)
+                    return true;
+                // Verify it is a proper subdomain such as bar.foo.com,
+                // not just a suffix of a domain such as bazfoo.com.
+                return uriDomain.charAt(beforeMatch) == '.';
+            }
+            return false;
+        }
+
+        private static boolean pathMatches(String path, String cookiePath)
+        {
+            if (cookiePath == null)
+                return true;
+            // RFC 6265, section 5.1.4, path matching algorithm.
+            if (path.equals(cookiePath))
+                return true;
+            if (path.startsWith(cookiePath))
+                return cookiePath.endsWith("/") || path.charAt(cookiePath.length()) == '/';
+            return false;
+        }
+
+        @Override
+        public boolean remove(URI uri, HttpCookie cookie)
+        {
+            Key key = new Key(uri.getScheme(), uri.getHost());
+            try (AutoLock ignored = lock.lock())
+            {
+                boolean[] result = new boolean[1];
+                cookies.compute(key, (k, v) ->
+                {
+                    if (v == null)
+                        return null;
+                    boolean removed = v.remove(cookie);
+                    result[0] = removed;
+                    return v.isEmpty() ? null : v;
+                });
+                return result[0];
+            }
+        }
+
+        @Override
+        public boolean clear()
+        {
+            try (AutoLock ignored = lock.lock())
+            {
+                if (cookies.isEmpty())
+                    return false;
+                cookies.clear();
+                return true;
+            }
+        }
+
+        private record Key(String scheme, String domain)
+        {
+            private Key(String scheme, String domain)
+            {
+                this.scheme = scheme;
+                this.domain = domain.toLowerCase(Locale.ENGLISH);
+            }
+        }
+
+        private static class Cookie extends HttpCookie.Wrapper
```

Comment:
perhaps:
```suggestion
        private static class SetCookie extends HttpCookie.Wrapper
```

Reply:
Not `SetCookie`, perhaps `StoredCookie`, but it's a private internal class, so I thought shorter was better.